### PR TITLE
Add SNI with DTLS tests to ssl-opt.sh script

### PR DESCRIPTION
**Pull request by andresag01**
This PR adds 3 tests for SNI with DTLS to the script ssl-opt.sh.

The change is related to IOTSSL-1114.

### CODE REVIEW

**Comment by andresag01** [comment]
@sbutcher-arm

File: tests/ssl-opt.sh [hunk 1]
```diff
@@ -2070,6 +2070,54 @@ run_test    "SNI: CA override with CRL" \
             -S "! The certificate is not correctly signed by the trusted CA" \
             -s "The certificate has been revoked (is on a CRL)"
 
+# Tests for SNI and DTLS
+
+run_test    "SNI/DTLS: matching cert 1" \
```
Comment (gilles-peskine-arm) [inline tests/ssl-opt.sh]: For consistency, the description should be `SNI: DTLS, matching cert 1`. Not a blocker.

Comment (andresag01) [reply tests/ssl-opt.sh]: I have renamed the tests for consistency.

**Comment by andresag01** [comment]
@gilles-peskine-arm: The initial 3 tests in SNI/DTLS are copies of the original SNI/TLS tests. There were only 3 because the idea was to do a quick check rather than duplicate the whole tests as the SNI functionality is mostly shared. I have now copied the missing SNI/DTLS into SNI/TLS.

**Comment by andresag01** [comment]
I have created backports for this PR:

* Mbed TLS 2.7: https://github.com/ARMmbed/mbedtls/pull/1664
* Mbed TLS 2.1: https://github.com/ARMmbed/mbedtls/pull/1665

**Comment by gilles-peskine-arm** [comment]
retest

**Comment by gilles-peskine-arm** [comment]
Marked “ready for merge”. We need new Jenkins results because the logs of the old run have been deleted but the backports pass and it's been tested manually so it's good enough to merge to -proposed at this point.

**Pull request merged**